src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1591,39 +1591,42 @@ void validateParamsType(BindingDiagnosticBag diagnostics)
                                 checkIsAtLeastAsVisible(syntax, binder, constructor, diagnostics);
                             }
 
-                            if (!binder.HasCollectionExpressionApplicableAddMethod(syntax, Type, out ImmutableArray<MethodSymbol> addMethods, diagnostics))
+                            if (binder.GetCollectionExpressionApplicableIndexer(syntax, Type, elementTypeWithAnnotations.Type, BindingDiagnosticBag.Discarded) is null)
```

Comment:
>BindingDiagnosticBag.Discarded [](http://example.com/codeflow?start=127&length=30)

Could we be losing useful diagnostics here in the case where an indexer is found?

Reply:
Added PROTOTYPE comment.